octobot_trading/personal_data/positions/states/idle_position_state.py

```diff
@@ -63,19 +71,24 @@ async def _synchronize_with_exchange(self, force_synchronization: bool = False)
         # Disable open position synchronization for now, let position updater refresh positions
         pass
 
-    async def terminate(self):
+    def sync_terminate(self):
         """
-        Should wait for being replaced by a LiquidatePositionState
+        Should wait for being replaced by an ActivePositionState or LiquidatePositionState
         """
         if not self.has_terminated:
             self.log_event_message(enums.StatesMessages.OPEN)
 
-            # set position mark price
-            self.position.mark_price = decimal.Decimal(
-                self.position.exchange_manager.exchange_symbols_data.
-                    get_exchange_symbol_data(self.position.symbol).prices_manager.mark_price)
+            if not self.position.mark_price:
```

Comment:
I think we did that to ensure that the mark_price is updated before changing state. I don't know if we still need it or not

Reply:
I removed it because it was using an old mark price on state change after mark price update 😂

Reply:
in this case position.mark_price just got updated